extensions/notebook/src/test/book/book.test.ts

```diff
@@ -34,13 +34,13 @@ export interface IExpectedBookItem {
 }
 
 export function equalBookItems(book: BookTreeItem, expectedBook: IExpectedBookItem, errorMsg?: string): void {
-	should(book.title).equal(expectedBook.title, `Book titles do not match, expected ${expectedBook?.title} and got ${book?.title}`);
-	if (expectedBook.file) {
+	should(book?.title).equal(expectedBook?.title, `Book titles do not match, expected ${expectedBook?.title} and got ${book?.title}`);
```

Comment:
Why would book be undefined? 

If that's actually a valid case then you should be updating the parameters to be `BookTreeItem | undefined` - but it seems odd to have either of these be possibly undefined

Reply:
I forgot to undo these changes once I fixed the actual issue. But I'll make the book to be BookTreeItem | undefined. Because the tests actual check the equality of the results and sometimes due to test failure the result can be undefined.

Reply:
I looked over the code and it seems like we're always guaranteeing that by the time this function is called we'll either have failed the test (due to undefined/length check) or have a valid BookTreeItem. So I don't think having this be undefined is necessary here.

Reply:
So the test failures Cory was talking about yesterday is because the book was undefined. It could happen again since the revealActiveDocumentInViewlet test just calls the method and checks the returned value directly.